src/cpp/library/CUDA/cuda_runtime_api.h

```diff
@@ -537,37 +559,48 @@ cudaError_t cudaThreadSynchronize()
     free(node);
   }
   lastError = CUDA_SUCCESS;
+  __ESBMC_atomic_end();
   return CUDA_SUCCESS;
 }
 
 extern __device__ __cudart_builtin__ cudaError_t CUDARTAPI
 cudaDeviceGetAttribute(int *value, enum cudaDeviceAttr attr, int device)
 {
+  __ESBMC_atomic_begin();
   lastError = CUDA_SUCCESS;
+  __ESBMC_atomic_end();
```

Comment:
Does an atomic block around these single assignments change the number of interleavings?